### DISCUSSION TITLE
AOS & LOS proper usage question

### DISCUSSION THREAD

**Discussion by rbureau85** [RocketCommunicationsInc/astro#131] [Q&A] [answered]
Hello again,
The documentation for AOS and LOS says you can use the military time designations. I have tried using "D", "d", "Delta" etc but none of those work. They don't appear to work in the storybook either. Can you assist on what I am missing?

Thanks
Renee, SSC

**Comment by markacianfrani** [comment]
Hey Renee, 

The timezone property doesn't currently affect the AOS/LOS timestamps. I think our (wrong) assumption was that these would already be formatted to your particular timezone, either from your data source or in your client code since you're passing them in directly. 

We'll go ahead and add this as a new feature now though. I'll follow up here as soon as that's released.

**Reply by markacianfrani** [reply]
Sorry, I was totally wrong here. Timezone does affect timestamps, but it looks like there's a bug when you dynamically change the timezone. https://codesandbox.io/s/infallible-cache-igvsx?file=/src/App.js

Hotfix incoming! Are you changing the timezone after render?

**Reply by rbureau85** [reply]
That might be a setting options we incorporate later on but right now we are just looking to track a few time zones in the header. I tried using 2 RuxClocks but that looks like a mess and does not fit in the header without css :root adjustments

**Comment by rbureau85** [comment]
Mark,
So I am trying to get AOS and LOS to be 2 different timestamps based on the
timezone of the RuxClock. Like behind 4 hours or ahead 6 hours to track
different time zones. I have not been able to get the AOS or LOS to
display anything useful. Even the canvas doesn't seem to be working based
on the documentation.

Example:
Clock is on MST (mountain time)
AOS = Dubi time
LOS = Hawaii time

Based on the documentation I thought passing in the military time letter
would adjust the aos/los but it shows nothing or 00:00:00.

Thanks!
Renee

**Comment by markacianfrani** [comment]
Gotcha! Are you able to share a snippet of your Clock implementation? Or just the values you're passing in?

**Reply by rbureau85** [reply]
![Screenshot from 2021-11-29 13-16-00](https://user-images.githubusercontent.com/23664388/143936614-1dc92c13-bfe9-49bb-ba31-a0a5a5d69a85.png)
![Screenshot from 2021-11-29 13-16-14](https://user-images.githubusercontent.com/23664388/143936617-df62e96c-58b7-493c-9335-ef2710452ce2.png)
![Screenshot from 2021-11-29 13-17-47](https://user-images.githubusercontent.com/23664388/143936814-fd78e2b4-3d11-4de2-8193-5a19377d5a6a.png)
![Screenshot from 2021-11-29 13-17-58](https://user-images.githubusercontent.com/23664388/143936818-c723bebd-cafd-455e-a323-61b6eea4542d.png)
![Screenshot from 2021-11-29 13-19-51](https://user-images.githubusercontent.com/23664388/143937107-132b1b69-c226-4e94-8f14-e2e80556f03a.png)
![Screenshot from 2021-11-29 13-20-01](https://user-images.githubusercontent.com/23664388/143937109-6c0eb8e6-b48d-4124-86e7-a7251a0cb022.png)

If I use strings with the property for timezones it gives me no aos or los. If I use numbers for Aos/los  the whole clock disappears, but if you just do AOS with aos={4} I get an aos that is ahead 4 hrs but does not continue to change with clock

**Reply by rbureau85** [reply]
Looks like the AOS is stuck on the top of the hour too. It did not include the minutes passed in to move ahead 4 hrs. I think there are more bugs in here than I originally thought :-(

**Reply by markacianfrani** [reply]
Ahh okay, so the AOS/LOS properties are meant to be static timestamps. It sounds like what you want are actually three separate Clocks. Something like https://codesandbox.io/s/agitated-panka-9nu07?file=/src/App.js ?

**Reply by rbureau85** [reply]
![Screenshot from 2021-11-29 13-49-55](https://user-images.githubusercontent.com/23664388/143941042-a72b1577-8182-4862-9d3c-3e421d9d6de5.png)
The "D" is still throwing and error so the military string props

Why are they static? What was the original purpose? 

Yes, I was going to use 3 different clocks but they take up a lot of space and having the AOS/LOS as other time zones shows connection between that main clock and the other needed zones. I would recommend that be a feature. As you know tracking of multiple time zones is a big thing in mission.

thanks for your time in explaining.

**Reply by markacianfrani** [reply]
AOS (Acquisition of Signal) and LOS (Loss of Signal) are usage specific and represent the specific times a signal was received and lost from a satellite. 

As for the spacing issue, you can control the margin of each clock (I updated the example above to show that) so essentially it can take up the same amount of space as using AOS/LOS would. We're currently working on design tokens that would give you consistent spacing variables for scenarios like this, but we will definitely bring this up internally and iterate on it. 

That invalid time error will be fixed shortly in a patch

**Reply by rbureau85** [reply]
Excellent, Thank you for the clarification. I definitely had it wrong in my
head.

**Comment by rbureau85** [comment, marked as answer]
> Ahh okay, so the AOS/LOS properties are meant to be static timestamps. It sounds like what you want are actually three separate Clocks. Something like https://codesandbox.io/s/agitated-panka-9nu07?file=/src/App.js ?

Current answer to this question is use 3 different clocks and know that AOS/LOS are static and after some digging in the source code require a string of something like 1995-12-17T03:24:00. The date part needs to be there but is irreverent to what is displayed. Your hour and mins will be displayed as what you pass in not difference from timezone selected.

**Comment by markacianfrani** [comment]
Just to close the loop here, we just released 6.0.4 which should fix that console error you were getting
